awx/main/expect/isolated_manager.py

```diff
@@ -20,143 +19,83 @@
-                 idle_timeout=None):
+    def __init__(self, cancelled_callback=None, idle_timeout=None):
         """
-        :param env:                 a dict containing environment variables for the
-                                    subprocess, ala `os.environ`
         :param cancelled_callback:  a callable - which returns `True` or `False`
                                     - signifying if the job has been prematurely
                                       cancelled
-        :param job_timeout          a timeout (in seconds); if the total job runtime
-                                    exceeds this, the process will be killed
         :param idle_timeout         a timeout (in seconds); if new output is not
                                     sent to stdout in this interval, the process
                                     will be terminated
         """
-        self.management_env = self._base_management_env()
         self.cancelled_callback = cancelled_callback
-        self.job_timeout = job_timeout
-        self.idle_timeout = idle_timeout
+        self.idle_timeout = idle_timeout or max(60, 2 * settings.AWX_ISOLATED_CONNECTION_TIMEOUT)
         self.started_at = None
 
-    @staticmethod
-    def _base_management_env():
-        '''
-        Returns environment variables to use when running a playbook
-        that manages the isolated instance.
-        Use of normal job callback and other such configurations are avoided.
-        '''
+    def build_runner_params(self, hosts, verbosity=1):
         env = dict(os.environ.items())
         env['ANSIBLE_RETRY_FILES_ENABLED'] = 'False'
         env['ANSIBLE_HOST_KEY_CHECKING'] = 'False'
         env['ANSIBLE_LIBRARY'] = os.path.join(os.path.dirname(awx.__file__), 'plugins', 'isolated')
-        return env
-
-    @staticmethod
-    def _build_args(playbook, hosts, extra_vars=None):
-        '''
-        Returns list of Ansible CLI command arguments for a management task
-
-        :param playbook:   name of the playbook to run
-        :param hosts:      host pattern to operate on, ex. "localhost,"
-        :param extra_vars: optional dictionary of extra_vars to apply
-        '''
-        args = [
-            'ansible-playbook',
-            playbook,
-            '-u', settings.AWX_ISOLATED_USERNAME,
-            '-T', str(settings.AWX_ISOLATED_CONNECTION_TIMEOUT),
-            '-i', hosts
-        ]
-        if extra_vars:
-            args.extend(['-e', json.dumps(extra_vars)])
-        if settings.AWX_ISOLATED_VERBOSITY:
-            args.append('-%s' % ('v' * min(5, settings.AWX_ISOLATED_VERBOSITY)))
-        return args
 
-    @classmethod
-    def awx_playbook_path(cls):
-        return os.path.abspath(os.path.join(
-            os.path.dirname(awx.__file__),
-            'playbooks'
-        ))
+        def finished_callback(runner_obj):
+            if runner_obj.status == 'failed':
+                stdout = runner_obj.stdout.read()
```

Comment:
@AlanCoding if a management playbook (any of them) fails, do `playbook_logger.error(stdout)` and also send a verbose event so we can see the failure at the `stdout` endpoint.

Reply:
We did this same thing before moving to runner (removed line 128). It's somewhat gory to get management playbook details in the job stdout, but when it can't be started, that instant feedback is really important.

The only difference I see is that the EOF event isn't sent, which I assume runner does. So this looks pretty good for just keeping on doing what it used to do.

Reply:
@AlanCoding I talked to @chrismeyersfsu and decided to remove the verbose event because we're worried it's going to mess w/ the event counting `event_processing_finished` mechanism.  If these playbooks fail, check the logs for error info.